### PR TITLE
Feat/client/search-filters/price-category

### PR DESCRIPTION
**Pull request by DLi7077**
Using @manav809 's price category filter
- Tweaked the styles a bit
- Moved the state to outside the component
- Integrate with server filter-supported query route

Lil demo showing the working filter

https://user-images.githubusercontent.com/51217000/236116607-450072da-5956-4ee1-87d6-d155cbd2f539.mp4

### CODE REVIEW

File: client/src/components/Filters/FoodCategoryFilter.js
Comment (DLi7077) [inline client/src/components/Filters/FoodCategoryFilter.js:1]: dont look

File: client/src/components/Filters/PriceFilter.js
Comment (DLi7077) [inline client/src/components/Filters/PriceFilter.js:1]: Manav's Implementation of price filter

File: client/src/pages/Search/index.js
Comment (DLi7077) [inline client/src/pages/Search/index.js:1]: renamed our localStorage variable `budget-eats-cache` to `query`

File: client/src/pages/Search/index.js [hunk 1]
```diff
@@ -68,6 +79,8 @@ export default function Search() {
     palette: {
       primary: {
         main: "hsl(25,80%,50%)",
+        chip: "hsl(25,85%, 57.5%)",
+        bland: "hsl(25,85%, 95%)",
```
Comment (DLi7077) [inline client/src/pages/Search/index.js:83]: prob wont need this after I start implementing the category filter

File: client/src/pages/Search/index.js [hunk 2]
```diff
@@ -77,6 +90,11 @@ export default function Search() {
     },
   });
 
+  function saveStateToLocalStorage() {
+    localStorage.setItem("query", JSON.stringify(query));
+    localStorage.setItem("price_categories", JSON.stringify(priceFilter));
+  }
```
Comment (DLi7077) [inline client/src/pages/Search/index.js:96]: Util to reduce lines of code

File: client/src/pages/Search/index.js [hunk 3]
```diff
@@ -123,6 +138,8 @@ export default function Search() {
           <SearchHeader
             updateFields={updateFields}
             searchFields={searchFields}
+            priceFilter={priceFilter}
+            setPriceFilter={setPriceFilter}
```
Comment (DLi7077) [inline client/src/pages/Search/index.js:142]: prop drilling price props into the filter component
Search/index.js -> SearchHeader -> PriceFilter

**Review by Remos96 — APPROVED** [review]
looks good

**Pull request merged**